src/runtime/ipx.ts

```diff
@@ -1,19 +1,116 @@
 import { fileURLToPath } from 'node:url'
-import { createIPX, createIPXH3Handler, ipxFSStorage, ipxHttpStorage } from 'ipx'
-import type { IPXOptions } from 'ipx'
+import { createIPX, createIPXH3Handler, ipxHttpStorage, unstorageToIPXStorage as originalUnstorageToIPXStorage } from 'ipx'
+import type { IPXOptions, IPXStorage, IPXStorageMeta } from 'ipx'
 import { lazyEventHandler, useBase } from 'h3'
 import { isAbsolute } from 'pathe'
 import type { NitroRuntimeConfig } from 'nitropack'
+import { createStorage, defineDriver } from 'unstorage'
+import fsLiteDriver from 'unstorage/drivers/fs-lite'
+import type { IPXRuntimeConfig } from '../ipx'
 import { useRuntimeConfig } from '#imports'
 
+// Storage and Driver are not exported from unstorage
+type StorageOrDriver = Parameters<typeof originalUnstorageToIPXStorage>[0]
+
+// Copied from ipx source, removed prefix and fixed to work with unstorage keys
+// Ref: https://github.com/unjs/ipx/blob/main/src/storage/unstorage.ts
+export function unstorageToIPXStorage (
```

Comment:
This is nice but I'm wondering we could directly support multiple dirs for IPX to swerve  to avoid all this logic that could also be potentially out of sync.

Reply:
(ipx does have it's own fs driver btw)

Reply:
- `unstorageToIPXStorage` needs to be fixed in IPX
- `ipxFSStorage` can be made to support multiple directories

Both are _better_ solutions. `unstorageToIPXStorage` needs to be fixed, as the current implementation does not work. However, main focus of this PR is squashing a bug from attached issue. Layers are niche use case, I think it should be handled locally.

When deciding to externalise solution, I propose considering `multi-fs` driver for `unstorage` instead of baking another driver into `ipx`.

Reply:
We shall simply support array for current IPX fs-provider. unstorage support for IPX is experimental only.

Reply:
How do you see the support for multiple dirs? Similar to how it's done with `combineUnstorages` here? Attempt to read file from every dir in series?

Reply:
Exactly. We shall iterate over (sanetized) dirs simply for when trying to load/stat a file instead of checking one probably also with a refactor to simplify stat check in a util.